tools/RepositoryManagement/DevHome.RepositoryManagement/ViewModels/RepositoryManagementItemViewModel.cs

```diff
@@ -2,61 +2,451 @@
+
+        set => _latestCommit = value ?? string.Empty;
+    }
+
+    private string _branch;
+
+    /// <summary>
+    /// Gets or sets the local branch name.  Nulls are converted to string.empty.
+    /// </summary>
+    public string Branch
+    {
+        get => _branch ?? string.Empty;
+        set => _branch = value ?? string.Empty;
+    }
 
     public bool IsHiddenFromPage { get; set; }
 
+    public bool HasAConfigurationFile { get; set; }
+
     [RelayCommand]
-    public void OpenInFileExplorer()
+    public async Task OpenInFileExplorer()
     {
-        throw new NotImplementedException();
+        await CheckCloneLocationNotifyUserIfNotFound();
+        OpenRepositoryInFileExplorer(RepositoryName, ClonePath, nameof(OpenInFileExplorer));
     }
 
     [RelayCommand]
-    public void OpenInCMD()
+    public async Task OpenInCMD()
     {
-        throw new NotImplementedException();
+        await CheckCloneLocationNotifyUserIfNotFound();
+        OpenRepositoryinCMD(RepositoryName, ClonePath, nameof(OpenInCMD));
     }
 
     [RelayCommand]
-    public void MoveRepository()
+    public async Task MoveRepository()
     {
-        throw new NotImplementedException();
+        // TODO: Save to the database before moving the folder.
+        var newLocation = await PickNewLocationForRepositoryAsync();
+
+        if (string.IsNullOrEmpty(newLocation))
+        {
+            _log.Information("The path from the folder picker is either null or empty.  Not updating the clone path");
+            return;
+        }
+
+        if (string.Equals(Path.GetFullPath(newLocation), Path.GetFullPath(ClonePath), StringComparison.OrdinalIgnoreCase))
+        {
+            _log.Information("The selected path is the same as the current path.  Not updating the clone path");
+            return;
+        }
+
+        var newDirectoryInfo = new DirectoryInfo(Path.Join(newLocation, RepositoryName));
+        var currentDirectoryInfo = new DirectoryInfo(Path.GetFullPath(ClonePath));
+
+        try
+        {
+            currentDirectoryInfo.MoveTo(newDirectoryInfo.FullName);
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Cound not move repository to the selected location.");
+            TelemetryFactory.Get<ITelemetry>().Log(
+                EventName,
+                LogLevel.Critical,
+                new RepositoryLineItemEvent(nameof(MoveRepository), RepositoryName));
+        }
+
+        var repository = GetRepositoryReportIfNull(nameof(MoveRepository));
+        if (repository == null)
+        {
+            return;
+        }
+
+        var didUpdate = _dataAccess.UpdateCloneLocation(repository, newDirectoryInfo.FullName);
+
+        if (!didUpdate)
+        {
+            _log.Warning($"Could not update the database.  Check logs");
+        }
+
+        ClonePath = Path.Join(newLocation, RepositoryName);
     }
 
     [RelayCommand]
-    public void DeleteRepository()
+    public async Task DeleteRepository()
```

Comment:
hmm, perhaps we should allow the user to undo the deletion somehow in the future

Reply:
OOhhh.  Like a buffer period.  I like it.